core/system.h

```diff
@@ -19,7 +20,21 @@ namespace dronecore {
 class DroneCoreImpl;
 class PluginImplBase;
 
-class Device
+
+/**
```

Comment:
1. This feels very odd sitting up here in the System class docs. 
2. Why is it called `Self`? What is implied by calling it Self? I hate this name - bit close to *self*.
3. What is it for? Why does a user need to know about this?

Reply:
>This feels very odd sitting up here in the System class docs.

Thanks for pointing this. I have created an for a discussion on this: https://github.com/dronecore/DroneCore/issues/325. I added docs because this interface is publicly accessible to apps.

>Why is it called Self? What is implied by calling it Self? I hate this name - bit close to self.
>What is it for? Why does a user need to know about this?

As mentioned in the header, this represents GCS DroneCore client. I mentioned in detail here: https://github.com/dronecore/DroneCore/issues/317 Its for plugins while sending MAVLink messages. We don't have class `SystemImpl` to add it in :)

Reply:
@hamishwillee what name do you suggest after you read through #317  ?

Reply:
I'm a bit rushed, but I saw another you renamed it  `struct ControlSystem` and that is much less offensive :-). Why not `GCSClient`?
Also still feels odd sitting in system.h alongside System, the object for our System.

Reply:
Yeah, `GCSClient` is better, I will change it. Its in [mavlink_system.h](https://github.com/dronecore/DroneCore/pull/310/files#diff-e3907820f42d7a290a4876d951b4ca80R25) not `system.h`.